packages/styled-components/src/sheet/dom.js

```diff
@@ -20,15 +20,16 @@ const findLastStyleTag = (target: HTMLElement): void | HTMLStyleElement => {
 };
 
 /** Create a style element inside `target` or <head> after the last */
-export const makeStyleTag = (target?: HTMLElement): HTMLStyleElement => {
+export const makeStyleTag = (useCSSOM: boolean, target?: HTMLElement): HTMLStyleElement => {
   const head = ((document.head: any): HTMLElement);
   const parent = target || head;
   const style = document.createElement('style');
   const prevStyle = findLastStyleTag(parent);
   const nextSibling = prevStyle !== undefined ? prevStyle.nextSibling : null;
 
   style.setAttribute(SC_ATTR, SC_ATTR_ACTIVE);
-  style.setAttribute(SC_VERSION_ATTR, SC_VERSION);
+  style.setAttribute(SC_ATTR_VERSION, SC_VERSION);
+  style.setAttribute(SC_ATTR_MODE, useCSSOM ? 'cssom' : 'text');
```

Comment:
Do we really need this tag in production?

Reply:
Nope, I'll remove it. Just something I thought might be useful down the line if more injection modes end up happening for debugging purposes.